compositor_render/src/transformations/web_renderer/node.rs

```diff
@@ -0,0 +1,62 @@
+use std::{collections::HashMap, sync::Arc};
+
+use compositor_common::scene::{NodeId, Resolution};
+use log::error;
+
+use crate::renderer::{
+    texture::{utils::pad_to_256, NodeTexture},
+    RenderCtx,
+};
+
+use super::WebRenderer;
+
+pub struct WebRendererNode {
+    renderer: Arc<WebRenderer>,
+    buffers: HashMap<NodeId, Arc<wgpu::Buffer>>,
+}
+
+impl WebRendererNode {
+    pub fn new(renderer: Arc<WebRenderer>) -> Self {
+        Self {
+            renderer,
+            buffers: HashMap::new(),
+        }
+    }
+
+    pub fn render(
+        &mut self,
+        ctx: &mut RenderCtx,
+        sources: &[(&NodeId, &NodeTexture)],
+        target: &mut NodeTexture,
+    ) {
+        for (id, texture) in sources {
+            let Some(texture_state) = texture.state() else {
+                continue;
+            };
+
+            let texture = texture_state.rgba_texture();
```

Comment:
You could move the code that makes sure the buffer is the correct size to RGBATexture code, or maybe to separate function in WebRendererNode